src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlArrayMethodCallTranslator.cs

```diff
@@ -0,0 +1,408 @@
+                return null;
+
+            switch (expression.Method.Name)
+            {
+            case "get_Item":
+                return MakeIndex(instance, expression.Arguments[0]);
+
+            case nameof(IList.IndexOf) when VersionAtLeast(9, 5):
+                return
+                    Expression.Subtract(
+                        Expression.Coalesce(
+                            new SqlFunctionExpression(
+                                "array_position",
+                                typeof(int?),
+                                new[] { instance, expression.Arguments[0] }),
+                            Expression.Constant(0)),
+                        Expression.Constant(1));
+
+            default:
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Translates instance methods defined on <see cref="string"/>.
+        /// </summary>
+        /// <param name="expression">The expression to translate.</param>
+        /// <returns>
+        /// A translated expression or null if no translation is supported.
+        /// </returns>
+        [CanBeNull]
+        static Expression StringInstanceHandler([NotNull] MethodCallExpression expression)
+        {
+            var instance = expression.Object;
+
+            if (instance == null || instance.Type != typeof(string))
+                return null;
+
+            switch (expression.Method.Name)
+            {
+            case "get_Chars":
+                return MakeIndex(instance, expression.Arguments[0]);
+
+            default:
+                return null;
+            }
+        }
+
+        #endregion
+
+        #region Helpers
+
+        /// <summary>
+        /// True if <see cref="_postgresVersion"/> is null, greater than, or equal to the specified version.
+        /// </summary>
+        /// <param name="major">The major version.</param>
+        /// <param name="minor">The minor version.</param>
+        /// <returns>
+        /// True if <see cref="_postgresVersion"/> is null, greater than, or equal to the specified version.
+        /// </returns>
+        bool VersionAtLeast(int major, int minor) => _postgresVersion is null || new Version(major, minor) <= _postgresVersion;
+
+        /// <summary>
+        /// Creates an expression of type <see cref="T:ExpressionType.ArrayIndex"/> or <see cref="T:ExpressionType.Index"/>
+        /// from the <paramref name="instance"/> and the <paramref name="index"/>.
+        /// </summary>
+        /// <param name="instance">The instance to index.</param>
+        /// <param name="index">The index value.</param>
+        /// <returns>
+        /// An <see cref="T:ExpressionType.ArrayIndex"/> or <see cref="T:ExpressionType.Index"/>.
+        /// </returns>
+        [NotNull]
+        static Expression MakeIndex([NotNull] Expression instance, [NotNull] Expression index)
```

Comment:
I'd split to two methods, one for array and another for list (string isn't needed if we move it out to another translator).

Reply:
I'm just going to omit this altogether and inline the code for arrays/lists.

Reply:
Scratch that, still need the methods, but will split.